angular2-jwt.ts

```diff
@@ -109,7 +109,19 @@ export class AuthHttp {
       this.setGlobalHeaders(this.config.globalHeaders, providedOpts);
     }
 
-    newOptions = newOptions.merge(new RequestOptions(providedOpts));
+    let providedOptions = new RequestOptions(providedOpts);
+    if (newOptions.headers && newOptions.headers.keys().length > 0) {
+      newOptions.headers.forEach((header: Object, name: string) => {
+
+        // set only not existing header keys
+        if (!providedOptions.headers.has(name)) {
+          let headerValue: string = (header as any)[Object.keys(header)[0]];
```

Comment:
Is this taking only the first entry of header? You know that you can have multiple headers of the same type right? Technically you should always be adding headers, regardless of if they're already set or not.